docs/book/src/roadmap.md

```diff
@@ -0,0 +1,42 @@
+# Cluster API Roadmap
+
+This roadmap is a constant work in progress, subject to frequent revision. Dates are approximations.
+
+## Ongoing
+
+- Documentation improvements
+
+## v0.3 (v1alpha3) ~ March 2020
+
+- Control plane provider concept, KubeadmControlPlane implementation
+- clusterctl v2 - easier installation & upgrades of Cluster API + providers
+- Integration with cloud provider VM scaling groups (MachinePools)
+- Machine failure remediation
+- Provider end to end testing framework
+
+## v0.4 (v1alpha4) ~ July 2020
+
+- [Machine bootstrap failure detection](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/972)
+- [Improved status conditions](https://github.com/kubernetes-sigs/cluster-api/issues/1658)
+- [Machine load balancers](https://github.com/kubernetes-sigs/cluster-api/issues/1250)
+- [Extensible machine pre-delete hooks](https://github.com/kubernetes-sigs/cluster-api/issues/1514)
+
+## v0.5 (v1???) ~ November 2020
+
+- ?
+
+## TBD
+
+- [Pluggable MachineDeployment upgrade strategies](https://github.com/kubernetes-sigs/cluster-api/issues/1754)
+- Dual stack IPv4/IPv6 support
+- Windows nodes
+- Auto-scaling support (either with cluster-autoscaler, or something else)
+- UEFI booting
+- GPU support (BitFusion, Elastic GPU, etc.)
+- SR-IOV support
```

Comment:
This seems like something that would be in the intersection of machine infrastructure providers and/or image-builder, rather than core cluster-api itself.

Reply:
Definitely could be. Is there anything generic about SR-IOV or is it all image & provider specific?

Reply:
Two ways:
Providers: Exposing the options to make use of  those features

Image:
Guest support for is cross-provider.
Also, on a modern enough Linux kernel, all the Amazon & Azure hardware is already supported these days.